.github/workflows/esp32-test.yaml

```diff
@@ -6,7 +6,15 @@
 
 name: ESP32 Test
 
-on: [push, pull_request]
+on:
+  push:
+    paths:
+      - 'src/platforms/esp32/**'
+      - 'src/libAtomVM/**'
+  pull_request:
+    paths:
+      - 'src/platforms/esp32/**'
+      - 'src/libAtomVM/**'
 
```

Comment:
We probably want to also include:
- .github/workflows
- CMakeLists.txt

Reply:
The workflows directory is needed here for sure, but I do not see where the top level CMakeLists.txt affects, or is involved with the esp32-test workflow... (am I missing something?) Unless we want to include it as a precautionary measure to future-proof changes to the build system.

Reply:
We definitely want `github/workflows` here, but I do not think the top level `CMakeLists.txt` is necessary here...